bin/update-wdtk-rejections

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# update-wdtk-rejections:
+#
+#  Run WDTK Rake tasks to generate requests to block at MTA layer and
+#  add updated file to Puppet.
+#
+
+WDTK_DIR=${1:-/data/vhost/www.whatdotheyknow.com/alaveteli}
+
+# Are we running as root?
+if [ "$UID" != "0" ]; then
+  echo "This script should be run as root, aborting." >&2
+  exit 1
+fi
+
+# Is WDTK available where we think it is?
+if [ ! -d "$WDTK_DIR" ]; then
+  echo "WDTK doesn't appear to be availle in ${WDTK_DIR}, aborting." >&2
+  exit 2
+fi
+
+check_rake () {
+  RAKE_TASK=$1
+  RAKE_EXIT_CODE=$2
+  if [ "${RAKE_EXIT_CODE}" != "0" ]; then
+    echo "Fatal: ${RAKE_TASK} rake task returned non-zero, aborting." >&2
+    exit 3
+  fi
+}
+
+RAKE_OUTPUT=$(mktemp)
+SORTED_OUTPUT=$(mktemp)
+
+# Update the list of requests to block
+su -l foi -c "cd ${WDTK_DIR} && bundle exec rake config_files:set_reject_incoming_at_mta REJECTED_THRESHOLD=3 AGE_IN_MONTHS=6 DRYRUN=0; exit $?" > /dev/null
```

Comment:
I assume the `exit $?` can't hurt, but just interested in is it necessary?
```
root# su -l matthew -c "cd /asdasdas;"; echo $?
-su: line 0: cd: /asdasdas: No such file or directory
1
```

Reply:
Yes, can't actually remember why I used this actually, as I know the exit code should be passed through.